### PR TITLE
SER-2686 | Publish wiki status changes

### PR DESCRIPTION
**Pull request by slayful**

### CODE REVIEW

File: includes/wikia/VariablesBase.php [hunk 1]
```diff
@@ -3914,6 +3914,11 @@
 	]
 ];
 
+$wgWikiStatusChangePublisher = [
```
Comment (mszabo-wikia) [inline includes/wikia/VariablesBase.php:3921]: Let's add some docs for this variable

File: extensions/wikia/WikiFactory/WikiStatusChangePublisher/WikiStatusChangeHooks.php [hunk 2]
```diff
@@ -0,0 +1,47 @@
+<?php
+
+use Wikia\Rabbit\ConnectionBase;
+
+/**
+ * @package MediaWiki
+ * @subpackage WikiStatusChangePublisher
+ *
+ */
+class WikiStatusChangeHooks {
```
Comment (mszabo-wikia) [inline extensions/wikia/WikiFactory/WikiStatusChangePublisher/WikiStatusChangeHooks.php:8]: This class needs to be registered with the autoloader

File: extensions/wikia/WikiFactory/WikiStatusChangePublisher/WikiStatusChangeHooks.php [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+	}
+
+	private static function getStatus( int $status ) {
+		switch ( $status ) {
+			case WikiFactory::CLOSE_ACTION:
+				return "closed";
+			case WikiFactory::PUBLIC_WIKI:
+				return "opened";
+			case WikiFactory::HIDE_ACTION:
+				return "hidden";
```
Comment (mszabo-wikia) [inline extensions/wikia/WikiFactory/WikiStatusChangePublisher/WikiStatusChangeHooks.php]: are there any other cases we should handle? should there be a fallback?

**Review by mszabo-wikia — CHANGES_REQUESTED** [review]

File: includes/wikia/rabbitmq/ConnectionBase.class.php [hunk 4]
```diff
@@ -16,12 +16,10 @@ class ConnectionBase {
 
 	protected $vhost;
 	protected $exchange;
-	protected $deadExchange;
```
Comment (slayful) [inline includes/wikia/rabbitmq/ConnectionBase.class.php:19]: This field was unused.

**Review by mszabo-wikia — APPROVED** [review]

**Review by mszabo-wikia — APPROVED** [review]

**Review by tmszdmsk — APPROVED** [review]

**Pull request merged**